sim/hunter/talents.go

```diff
@@ -61,7 +61,12 @@ func (hunter *Hunter) ApplyTalents() {
 	hunter.AddStat(stats.MeleeCrit, float64(hunter.Talents.KillerInstinct)*1*core.CritRatingPerCritChance)
 
 	if hunter.Talents.LethalShots > 0 {
-		hunter.AddStat(stats.MeleeCrit, 1*float64(hunter.Talents.LethalShots)*core.CritRatingPerCritChance)
+		for _, spell := range hunter.Shots {
+			if spell != nil {
+				spell.BonusCritRating += 1*float64(hunter.Talents.LethalShots)*core.CritRatingPerCritChance
+			}
+		}
+		hunter.AutoAttacks.RangedConfig().BonusCritRating += 1*float64(hunter.Talents.LethalShots)*core.CritRatingPerCritChance
```

Comment:
Small performance knit: You could save the bonus crit to a variable so that you don't re-calculate it for each ability

Reply:
done!